File: app-dispatcher-diamond/dispatcher-api/src/main/java/com/jay/dispatcher_api/IAppLife.java
package com.jay.dispatcher_api;

import android.content.Context;

/**
 * 组件只需实现该接口
 *
 * @author wangxuejie
 * @version 1.0
 * @date 2/18/21
 */
public interface IAppLife {

    
    int MAX_PRIORITY = 10;
    int MIN_PRIORITY = 1;
    int NORM_PRIORITY = 5;

    int getPriority();

    void onCreate(Context context);

    void onTerminate();


}

File: app-dispatcher-bronze/dispatcher-bronze/src/main/java/com/qlife/lib_app/appdelegate/IAppLife.kt
package com.qlife.lib_app.appdelegate

import android.app.Application
import android.content.Context
import android.content.res.Configuration

/**
 * @author wangxuejie
 * @version 1.0
 * @date 2020/3/31
 */
interface IAppLife {

    /**
     * 多进程会调用多次
     */
    fun attachBaseContext(context: Context, processName: String) {}

    /**
     * 多进程会调用多次
     */
    fun onCreate(application: Application, processName: String)

    fun onConfigurationChanged(newConfig: Configuration) {}

    fun onLowMemory() {}

    fun onTrimMemory(level: Int) {}

    fun onTerminate(application: Application) {}

    /**
     * 设置该appLife的优先级，必须设置，否则不会回调
     *
     */
    fun onPriority(): Int

    companion object {

        const val TAG = "AppLife"

    }

}
